enhancements/20220722-pci-passthrough.md

```diff
@@ -0,0 +1,219 @@
+the three NVIDIA devices by checking the box next to them. By default, all the nodes 
+with that device are selected. Then she presses the "Pass devices through on selected nodes"
+button.
+
+Alice goes to create a VM, and in the form to create a VM, there is a "PCI devices" list.
+When she selects a PCI Device, the Node Scheduling section is automatically updated to 
+"Run VM on node(s) matching scheduling rules", where the scheduling rule is "node X has 
+device D". This will ensure that the VM only runs on nodes with the selected device.
+
+The UI for selecting PCI devices will automatically narrow down to the intersection of all 
+the PCI devices belonging to the nodes that contain the already-selected devices. To show this 
+with an example, suppose Alice selected "NVIDIA GeForce GTX 1660", then, since only 
+node1 and node3 have that device, only the devices on node1 and node3 would be included 
+in the list. 
+
+node1: $\{\text{geforce1660}, \text{radeon6800}\}$
+
+node3: $\{\text{geforce1660}, \text{teslaV100}, \text{radeon6800}\}$
+
+And their intersection is: 
+
+$\{\text{geforce1660}, \text{radeon6800}\} \cap \{\text{geforce1660}, \text{teslaV100}, \text{radeon6800}\} = \{\text{geforce1660}, \text{radeon6800} \}$
+
+So only those two devices would remain in the PCI Devices list. As soon as any are unselected, the list would expand again.
+
+Once the VM is created, it is scheduled to run on the nodes with those devices, and those devices have already been configured 
+for PCI-Passthrough.
+
+
+#### Story 2
+Alice wants to run a DPDK Application in her guest cluster and 
+make use of 
+[SR-IOV](20220614-single-root-io-virtualization.md), 
+also set up in the guest cluster.
+
+First she has to enable PCI Passthrough so that the 
+[SR-IOV](20220614-single-root-io-virtualization.md)
+in the guest cluster can directly use 
+the underlying PF.
+    
+
+### User Experience In Detail
+
+#### Story 1 in detail
+Alice needs to have installed Harvester with the PCI devices plugged in. The installer detected
+the GPUs and installed the necessary drivers.
+
+After installing Harvester, it reboots. Since GPU drivers were installed, Harvester 
+also enabled the `iommu=on` kernel parameter. This protects the host memory 
+even though the guest has direct access to that memory.
+
+She enables PCI Passthrough by visiting the Advanced &rarr; PCI Passthrough interface 
+described in [story 1](#story-1). That list of devices and their nodes is stored as a json 
+object in the setting `pci-passthrough-config`:
+
+```json
+{
+    devices: [
+        {id: 1, name: "Intel Corporation Ethernet Connection (11) I219-LM"}
+    ],
+    nodes: [
+        {name: "node1", pciDevicesToPassThrough: [1]}
+    ]
+}
+```
+
+After that setting is updated, the settings controller will loop through the devices and prepare them 
+for PCI Passthrough:
+
+The preparation for passthrough consists of unbinding the driver, doing a driver override, then binding the 
```

Comment:
I'm not quite sure why we need a driver in the host if it's eventually unbound. By searching tutorials on Internet, most of them are just binding the device to vfio-pci.